In [1]:
import pandas as pd
import numpy as np
import scipy
from scipy import stats
from scipy.stats import norm

In [2]:
df = pd.read_csv('Costomer+Orderform.csv')

In [3]:
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [4]:
df.tail()

Phillippines   Indonesia       Malta       India
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

In [5]:
df.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [6]:
df.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [7]:
df.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [8]:
df.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

# Assumptions

Null hypothesis(H0):- customer order forms defective % does not varies by centre

Alternative hypothesis(H1):- customer order forms defective % varies by centre

In [10]:
#Making dimensional array

obs = np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [11]:
from scipy.stats import chi2_contingency

In [12]:
chi2_contingency(obs)

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

# Inferences

Since, p-value=0.277 is greater than α=0.05 we are rejecting Null hypothesis. Thus,customer order forms defective % does not varies by centre